### DISCUSSION TITLE
tensordot and einsum | benchmark

### DISCUSSION THREAD

**Discussion by ShHsLin** [NVIDIA/cuQuantum#110] [Q&A] [answered]
Hi team,

I would like to know if there is a benchmark of the functions, tensordot and einsum, against the implementation of other libraries, e.g. JAX, pytorch, .... 

I guess the cuQuantum implementation should definitely be faster, but I wonder whether this is tested anywhere. It can be of the similar tests like what Paul Springer did several years ago for [CPUs](
http://tensornetwork.org/benchmarks/), where tests are run over a collective of different order/shape tensors.

If such benchmark data exists and show prevailing performance against existing implementation in the libraries, e.g. [here](https://github.com/pytorch/pytorch/pull/84890), then there will be more motivation for integrating the cutensor/cuQuantum functions for other libraries. And of course, it will add a beautiful plot to the [cuQuantum SDK page](https://developer.nvidia.com/cuquantum-sdk).

**Comment by fmozafari** [comment, marked as answer]
Hi @ShHsLin,

Our team frequently conduct comprehensive benchmarking of cuTensorNet against other software packages commonly used in the Python community.  [Here](https://developer.nvidia.com/blog/accelerate-quantum-circuit-simulation-with-nvidia-cuquantum-23-10/) is some preliminary results comparing cuTensorNet with JAX and PyTorch that shows significant speedup in both finding contraction paths and executing actual tensor network contractions. We aim to share these results in the future. Moreover, you can use cuquantum-benchmarks for benchmarking. Stay tuned.
